src/codewhisperer/client/codewhisperer.ts

```diff
@@ -61,6 +68,20 @@ export class DefaultCodeWhispererClient {
                                 req.httpRequest.headers['x-amzn-codewhisperer-optout'] = `${isOptedOut}`
                             })
                         }
+
+                        // This logic is for backward compatability with legacy SDK v2 behavior for refreshing
+                        // credentials. Once the Toolkit adds a file watcher for credentials it won't be needed.
+                        if (isCloud9() && req.operation !== 'getAccessToken') {
+                            req.on('retry', resp => {
+                                if (
+                                    resp.error?.code === 'AccessDeniedException' &&
+                                    resp.error.message.match(/expired/i)
+                                ) {
+                                    refreshCredentials()
+                                    resp.error.retryable = true
```

Comment:
`resp.error.retryable = true`

will this change retry API call?

Reply:
Yeah it technically should.

Internally, requests are state machines, however, _a lot_ of behavior is determined by side-effects that happen due to events. Whether or not to retry an API call is one example of this. So even if I say the error is retryable, some other event listener can reset the flag back to false. Or if I don't set `retryable` in the right state, then the request is never retried anyway.